### PR TITLE
Reorganize actions doc.

### PR DESCRIPTION
**Pull request by rabbah**
This PR introduces a language-agnostic (mostly) basic introduction to working with actions. And it splits the language/runtime specific tutorials and references into their own markdowns. This is based on discussion with @csantanapr and @akrabat. All feedback appreciated and welcomed.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [x] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: docs/actions.md [hunk 1]
```diff
@@ -17,1182 +17,366 @@
...
-* [Go](#creating-go-actions)
-* [Native binaries](#creating-native-actions)
-
-Each invocation of an action results in an activation record that is identified by a unique
-activation ID. The input to an action and the result of an action are a dictionary of key-value
-pairs, where the key is a string and the value a valid JSON value.
-
-In addition, we recommend that you review the following topics:
+The rest of this document provides a [basic tutorial](#the-basics) for invoking an action, passing parameters to it, and
+inspecting the result. We recommend that you get familiar with the following topics first.
```
Comment (akrabat) [inline docs/actions.md]: We can't recommend that you get familiar with the following topics first (i.e. before the basic tutorial) when the first prerequisite topic item is the basic tutorial.

File: docs/actions.md [hunk 2]
```diff
@@ -1255,48 +437,64 @@ ok: got action hello
...
-https://${APIHOST}/api/v1/namespaces/${NAMESPACE}/actions/actionName
+https://${APIHOST}/api/v1/namespaces/${NAMESPACE}/actions/greeting
 ```
 
-For [web actions](https://github.com/apache/incubator-openwhisk/blob/master/docs/webactions.md#web-actions), a URL will be returned in the the following format:
+Authentication is required when invoking an action via an HTTPS request using this resource path.
+For more information regarding action invocations using the REST interface, see [Using REST APIs with OpenWhisk](rest_api.md#actions).
+
+Another way of invoking an action which does not require authentication is via
+[web actions](https://github.com/apache/incubator-openwhisk/blob/master/docs/webactions.md#web-actions).
```
Comment (akrabat) [inline docs/actions.md]: Can we use a relative link here?

**Comment by bdelacretaz** [comment]
I like it! I think it makes perfect sense to explain the general principles of Actions and farm out the language-specific details. It _might_ be useful to mention what happens under the hood when running actions (i.e. /init and /run calls to a container) but maybe that's too much detail at this level.

File: docs/php-actions.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+If your PHP action requires [Composer](https://getcomposer.org) dependencies, you can install them as usual using `composer require` which will create a `vendor` directory. Add this directory to your action's zip file and create the action:
+
+```bash
+zip -r helloPHP.zip index.php vendor
+wsk action create helloPHP --kind php:7.1 helloPHP.zip
+```
+
+The PHP runtime will automatically include Composer's autoloader for you, so you can immediately use the dependencies in your action code.
+
+Note that if you don't include your own `vendor` folder, then the runtime will include one for you. The packages included are listed in the [reference](https://github.com/apache/incubator-openwhisk/blob/master/docs/reference.md#composer-packages).
```
Comment (akrabat) [inline docs/php-actions.md]: Link is wrong - reference is now immediately below this sentence.

File: docs/swift-actions.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+You can create a OpenWhisk action called `helloSwift` from this function as
+follows:
+
+```
+wsk action create helloSwift hello.swift --kind swift:4.1
+```
+
+
+See the Swift [reference](./reference.md#swift-actions) for more information about the Swift runtime.
```
Comment (akrabat) [inline docs/swift-actions.md]: Link is wrong - reference is further down the page now.

File: docs/node-actions.md [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+
+2. Create an action from the following JavaScript function. For this example, the action is called 'hello'.
+
+  ```
+  wsk action create hello hello.js
+  ```
+  ```
+  ok: created action hello
+  ```
+  The CLI automatically infers the type of the action by using the source file extension. For `.js` source files, the action runs by using a Node.js 6 runtime. You can also create an action that runs with Node.js 8 by explicitly specifying the parameter `--kind nodejs:8`. For more information, see the Node.js 6 vs 8 [reference](./reference.md#javascript-runtime-environments).
```
Comment (akrabat) [inline docs/node-actions.md]: Link is wrong - reference is further down the page.

File: docs/node-actions.md [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+                  var temperature = condition.temp;
+                  var output = 'It is ' + temperature + ' degrees in ' + location + ' and ' + text;
+                  resolve({msg: output});
+              }
+          });
+      });
+  }
+  ```
+
+  Note that the action in the example uses the JavaScript `request` library to make an HTTP request to the Yahoo Weather API, and extracts fields from the JSON result. The [References](./reference.md#javascript-runtime-environments) detail the Node.js packages that you can use in your actions.
```
Comment (akrabat) [inline docs/node-actions.md]: Link to reference needs updating.

File: docs/docker-actions.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+  However, if there is a warm container using a previous version of your Docker image, any new invocations will continue to use that image unless you run `wsk action update`. This will indicate to the system that for new invocations it should execute a docker pull to get your new Docker image.
+
+  ```
+  ./buildAndPush.sh janesmith/blackboxdemo
+  ```
+  ```
+  wsk action update example --docker janesmith/blackboxdemo
+  ```
+
+  You can find more information about creating Docker actions in the [References](./reference.md#docker-actions) section.
```
Comment (akrabat) [inline docs/docker-actions.md]: The "Docker actions" section is no longer in reference.md, but I can't find it in docker-actions.md either!

File: docs/php-actions.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+PHP actions always consume an associative array and return an associative array. The entry method for the action is `main` by default but may be specified explicitly when creating the action with the `wsk` CLI using `--main`, as with any other action type.
+
+You can create an OpenWhisk action called `helloPHP` from this function as follows:
+
+```
+wsk action create helloPHP hello.php
+```
+
+The CLI automatically infers the type of the action from the source file extension. For `.php` source files, the action runs using a PHP 7.1 runtime. See the PHP [reference](./reference.md#php-actions) for more information.
```
Comment (akrabat) [inline docs/php-actions.md]: Link needs updating

**Comment by csantanapr** [comment]
This is great I wanted this split for the longest time +1

What do you think is all files start with “actions-*”? Like “actions-java.md”

I think is my OCD and I will have trouble find the files. What used to be in actions.md now is in actions-java.md also when listing the files all languages are together so it’s easy to see the ones available.

**Comment by rabbah** [comment]
hehe ok. I named them as I did for tab completion but I will rename.

**Comment by rabbah** [comment]
@bdelacretaz I added a commit to introduce init and run, cold vs warm start.

**Comment by rabbah** [comment]
@akrabat I've fixed the links and made a pass to update all the docs for the runtimes as well while at it.

File: docs/actions-php.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+```bash
+zip -r helloPHP.zip index.php vendor
+wsk action create helloPHP --kind php:7.1 helloPHP.zip
+```
+
+The PHP runtime will automatically include Composer's autoloader for you, so you can immediately
+use the dependencies in your action code.
+
+Note that if you don't include your own `vendor` folder, then the runtime will include one for you.
+The following PHP extensions are available in addition to the standard ones:
```
Comment (akrabat) [inline docs/actions-php.md]: Need to separate PHP extensions list from the Composer info as they aren't the same thing. Composer manages PHP components, PHP extensions are compiled C libraries that are loaded by the language and the the user cannot add to or remove them.

Comment (rabbah) [reply docs/actions-php.md]: thanks @akrabat for the review - I think addressed in my last commit if you can kindly look again.

Comment (akrabat) [reply docs/actions-php.md]: Yes. That's great. Thanks.

**Comment by csantanapr** [comment]
❤️ the move of references to each runtime language

**Pull request merged**